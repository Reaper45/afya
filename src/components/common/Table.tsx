import * as React from 'react'

import styled from "lib/emotion";

const TableWrapper = styled("table")``;

const TableHeader = styled("thead")``;

const TableHeaderCell = styled("th")``;

const TableBody = styled("tbody")``;

const TableRow = styled("tr")``;

const TableCell = styled("td")``;

const TableFooter = styled("tfoot")``;

interface TableTypeComponent extends React.FC {
  Body: typeof TableBody
  Cell: typeof TableCell
  Footer: typeof TableFooter
  Header: typeof TableHeader
  HeaderCell: typeof TableHeaderCell
  Row: typeof TableRow
}

/**
 * A table displays a collections of data grouped into rows.
 */
const Table: TableTypeComponent = ({ children }) => {
  return <TableWrapper>{children}</TableWrapper>;
};

Table.Header = TableHeader;

Table.HeaderCell = TableHeaderCell;
  
Table.Body = TableBody;

Table.Row = TableRow;

Table.Cell = TableCell;

Table.Footer = TableFooter;

export default Table ;
