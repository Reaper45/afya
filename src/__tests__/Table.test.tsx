import React from "react";
import { shallow } from "enzyme";

import Table from "components/common/Table";

it("renders table correctly", () => {
  const component = shallow(
    <Table>
      <Table.Header>
        <Table.HeaderCell>Id</Table.HeaderCell>
        <Table.HeaderCell>Name</Table.HeaderCell>
        <Table.HeaderCell>Description</Table.HeaderCell>
      </Table.Header>
      <Table.Body>
        <Table.Row>
          <Table.Cell>1</Table.Cell>
          <Table.Cell>Test Name</Table.Cell>
          <Table.Cell>Test Description</Table.Cell>
        </Table.Row>
      </Table.Body>
    </Table>
  );
  expect(component).toMatchSnapshot();
});
