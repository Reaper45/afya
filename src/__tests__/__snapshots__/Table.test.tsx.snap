// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table correctly 1`] = `
<Styled(table)>
  <Styled(thead)>
    <Styled(th)>
      Id
    </Styled(th)>
    <Styled(th)>
      Name
    </Styled(th)>
    <Styled(th)>
      Description
    </Styled(th)>
  </Styled(thead)>
  <Styled(tbody)>
    <Styled(tr)>
      <Styled(td)>
        1
      </Styled(td)>
      <Styled(td)>
        Test Name
      </Styled(td)>
      <Styled(td)>
        Test Description
      </Styled(td)>
    </Styled(tr)>
  </Styled(tbody)>
</Styled(table)>
`;
